### DISCUSSION TITLE
Error, pls help

### DISCUSSION THREAD

**Discussion by EmmanuelAjobo** [Cyfrin/foundry-full-course-cu#2409] [General]
i am trying to use zksync plugin, i am unable to deploy and also verify contract

Error: "underlying network changed (event=\"changed\", network={\"chainId\":324,\"name\":\"zkSync Era\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

![Screenshot 2024-09-06 092345](https://github.com/user-attachments/assets/723d53de-51eb-47a7-a40b-fd24b5940411)

**Comment by EngrPips** [comment]
Hello @Destructoe, Deploying with `zksync` can be difficult at times. You can retry later or skip deploying to `zksycn` as deploying yo `zksync` is optional. Don't sweat over it.

**Comment by Haardtripathi** [comment]
Hello @Destructoe , did you bridge your sepolia funds into zkSync Sepolia and then change network to zkSync on your metamask?
Tell me if you need more information.

**Reply by EmmanuelAjobo** [reply]
yes i have bridged it 
i have zksync on my wallet too

**Reply by Haardtripathi** [reply]
But did you change the network to zksync sepolia?
![image](https://github.com/user-attachments/assets/d0901e7d-a78e-4e6f-9724-4b25aef63f2d)

**Comment by TrustWarden** [comment]
change the network from sepolia to zkSync testnet in your wallet

**Reply by EmmanuelAjobo** [reply]
Yes i have done that too, didn't work out

**Reply by TrustWarden** [reply]
the error said that, idk what else is, did you put all files into contracts folder?

**Comment by EngrPips** [comment]
Interesting.

**Comment by 0xArDANT** [comment]
It looks like your wallet is connected on the Sepolia network instead of zSsync Sepolia.

**Reply by EmmanuelAjobo** [reply]
It is connected to zksync, i have done the bridging

**Reply by EmmanuelAjobo** [reply]
It is still not working

**Reply by EngrPips** [reply]
This part of the course is optional please don't sweat over it. You can come back to it later.

**Comment by 0xArDANT** [comment]
Network "Zksync Era" and "Chain ID: 324" are for Zksync mainnet.
The detected network Sepolia with chain ID 11155111 is Sepolia testnet on Ethereum. 

So your Zksync plugin is expecting the Zksync mainnet but detecting Ethereum Sepolia instead.


Verify that you configured your plugin well and imported the right Zksync Sepolia network on Metamask. Verify the chain ID on Chainlist website.

**Comment by EngrPips** [comment]
Interesting.

**Reply by 0xArDANT** [reply]
Let us know if it worked or not.
